test/scopedvalues.jl

```diff
@@ -126,3 +126,17 @@ end
         @test sval_float[] == 1.0
     end
 end
+
+# Test that the `@with` macro doesn't introduce unnecessary PhiC nodes
+# (which can be hard for the optimizer to remove).
+function with_macro_slot_cross()
+    a = 1
+    @with sval=>1 begin
+        a = sval_float[]
```

Comment:
This is intended to be `a = sval[]`? It's a minor issue, so I believe it can be addressed alongside other upcoming changes.

Reply:
No, this is as intended, the test is for the value of the slot `a` across the `@with` region and I didn't want it to fold